embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/JdbcInputConnection.java

```diff
@@ -351,6 +356,22 @@ public PreparedQuery wrapIncrementalQuery(String rawQuery, JdbcSchema querySchem
         return parameters.build();
     }
 
+    private List<JdbcLiteral> replacePlaceholder(StringBuilder sb, String rawQuery, JdbcSchema querySchema,
+                                                 List<Integer> incrementalColumnIndexes, List<JsonNode> incrementalValues)
+    {
+        ImmutableList.Builder<JdbcLiteral> parameters = ImmutableList.builder();
+        for (int n = 0; n < incrementalColumnIndexes.size(); n++) {
+            int columnIndex = incrementalColumnIndexes.get(n);
+            String columnName = querySchema.getColumnName(columnIndex);
+            JsonNode value = incrementalValues.get(n);
+            rawQuery = rawQuery.replaceAll(":" + columnName, "?");
```

Comment:
Sorry to be late.

> In some cases it won't work correctly.

I think using something SQL parser is too much in this case.
What do you think about replace with regular expression?

```java
rawQuery = rawQuery.replaceAll(":" + columnName + "[^\'\"]+?", " ? ");
```
At least, "replace longer column names previously" problem will be solved.
The pattern of regex is arguable and isn't a perfect solution, but.

> At least we should add annotation about it to README.

OK. I'll add it.

Reply:
> I think using something SQL parser is too much in this case.

I think so too.

> What do you think about replace with regular expression?

Replacing with regular expression is difficult because 
- DBMSs have various special characters. 
- column name might contain special characters (ex. `$`).

And I found that the method doesn't work correctly in the following cases.
- Same column name appears multiple times. Parameters will be short.
`... where :a > x or :a = x and :b > y`

- The order of column names in incremental_columns and SQL are different.
<pre>
incremental_columns : [a, b]
query: ... where :b > y and :a > x</pre>

How about the following alogrithm?

1. create map whose keys are the column names and whose values are JdbcLiteral objets.
2. sort the column names in the order of longer lengths.
3. find ':' from the beginning of the query string.
4. if the rest of the query string begins with one of the column names, replace with '?' and add JdbcLiteral object to parameters.
5. find ':' from the rest of the query string. -> 4

Reply:
OK. Let me implement with your suggestion.